service/src/main/java/uk/gov/hmcts/reform/fpl/service/ChildrenService.java

```diff
@@ -30,4 +39,38 @@ public String getChildrenLabel(List<Element<Child>> children) {
     public void addPageShowToCaseDetails(CaseDetails caseDetails, List<Element<Child>> children) {
         caseDetails.getData().put("pageShow", children.size() <= 1 ? "No" : "Yes");
     }
+
+    public List<Element<Child>> updateFinalOrderIssued(List<Element<Child>> children,
+        String orderAppliesToAllChildren, ChildSelector childSelector) {
+
+        if (YES.getValue().equals(orderAppliesToAllChildren)) {
+            return children.stream()
+                .map(child -> updateFinalOrderIssuedOnChild(child, YES))
+                .collect(Collectors.toList());
+        } else {
+            List<Integer> selectedChildren = childSelector != null ? childSelector.getSelected() : new ArrayList<>();
+            return IntStream.range(0, children.size())
+                .mapToObj(index -> {
+                    Element<Child> child = children.get(index);
+                    if (!selectedChildren.isEmpty() && selectedChildren.contains(index)) {
+                        return updateFinalOrderIssuedOnChild(child, YES);
+                    } else if (StringUtils.isEmpty(child.getValue().getParty().getFinalOrderIssued())) {
+                        return updateFinalOrderIssuedOnChild(child, NO);
```

Comment:
do we need this else if? can we always set no by default?

Reply:
This is to ensure that we do not override a flag that has been already set. I.e have 3 children on the case, 1st pass final order issued for child 1 only then child 1 - "Yes", child 2 - "No", child 3 - "No". 
2nd pass final order issued for child 2 only (need to ensure we do not set child 1 to "No" as they are not in the child selection list)  then  - child 1 - "Yes", child 2 - "Yes", child 3 - "No".